solr/core/src/java/org/apache/solr/cluster/placement/AttributeValues.java

```diff
@@ -65,11 +65,16 @@
   /**
    * For the given node: metric of specific name and registry
    */
-  Optional<Double> getMetric(Node node, String metricName, AttributeFetcher.NodeMetricRegistry registry);
+  Optional<Object> getMetric(Node node, String metricName, AttributeFetcher.NodeMetricRegistry registry);
```

Comment:
If `ReplicaMetrics` can be improved as suggested by a [comment](https://github.com/apache/lucene-solr/pull/2133/files#r546733523) there to make better use of `ReplicaMetric<T>` instances and not use generic `Object` values, can we generalize the concept?
Either rename `ReplicaMetric` and have it define metrics for all types of cluster entities (i.e. including nodes) or introduce a similar `NodeMetric<T>` class that would be used here as well as in `AttributeFetcher`?

Reply:
Also, method likely needs to have `Node` in the name given it's a node metric (like method below).